presto-main/src/test/java/com/facebook/presto/TestClientRequestFilterPlugin.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto;
+
+import com.facebook.airlift.http.server.Authenticator;
+import com.facebook.presto.server.MockHttpServletRequest;
+import com.facebook.presto.server.security.AuthenticationFilter;
+import com.facebook.presto.server.security.SecurityConfig;
+import com.facebook.presto.server.testing.TestingPrestoServer;
+import com.facebook.presto.spi.ClientRequestFilter;
+import com.facebook.presto.spi.ClientRequestFilterContext;
+import com.facebook.presto.spi.ClientRequestFilterFactory;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import javax.servlet.http.HttpServletRequest;
+
+import java.security.Principal;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.testng.Assert.assertEquals;
+
+public class TestClientRequestFilterPlugin
+{
+    static class GenericClientRequestFilterFactory
```

Comment:
Move test classes to the bottom.  Put fields of `TestClientRequestFilterPlugin` first, then tests, then helper methods, then inner classes.